### PR TITLE
Rename account to portfolio

### PR DESCRIPTION
**Pull request by msm-faraj**

### CODE REVIEW

**Comment by pouyapourbaba** [comment]
You can rebase this branch on your latest master locally and then push here so the changes from the previous PR are not here anymore. Take the following steps:

1. git checkout master
2. git pull
3. git checkout rename-account-to-portfolio
4. git rebase master
5. git push

File: src/controllers/portfolio/portfolio-controller.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const httpStatus = require("http-status");
+const { AuthenticationError, NotFoundError } = require("../../lib/error");
+
+class PortfolioController {
+  constructor({ User, Portfolio }) {
+    this.User = User;
+    this.Portfolio = Portfolio;
+  }
+
+  async createPortfolio(req, res) {
```
Comment (pouyapourbaba) [inline src/controllers/portfolio/portfolio-controller.js:10]:
```suggestion
  async create(req, res) {
```

File: src/controllers/portfolio/portfolio-controller.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      name,
+      description,
+      userId: req.authenticatedUser.id,
+      cash,
+      currency,
+    });
+    return res.status(httpStatus.CREATED).send(portfolio);
+  }
+
+  async findAllPortfolios(req, res) {
```
Comment (pouyapourbaba) [inline src/controllers/portfolio/portfolio-controller.js:28]:
```suggestion
  async findAll(req, res) {
```

File: src/controllers/portfolio/portfolio-controller.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    let portfolios = await this.Portfolio.findAll({
+      where: { deletedAt: null, userId: req.authenticatedUser.id },
+    });
+    return res.status(httpStatus.OK).send(portfolios);
+  }
+
+  async findOnePortfolio(req, res) {
```
Comment (pouyapourbaba) [inline src/controllers/portfolio/portfolio-controller.js:35]:
```suggestion
  async findOne(req, res) {
```

File: src/controllers/portfolio/portfolio-controller.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    const portfolio = await this.Portfolio.findOne({
+      where: { deletedAt: null, id: req.params.id },
+    });
+    if (!portfolio) throw new NotFoundError("Portfolio not found");
+
+    return res.status(httpStatus.OK).send(portfolio);
+  }
+
+  async updatePortfolio(req, res) {
```
Comment (pouyapourbaba) [inline src/controllers/portfolio/portfolio-controller.js:44]:
```suggestion
  async update(req, res) {
```

File: src/controllers/portfolio/portfolio-controller.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    const updatedPortfolio = await portfolio.update({
+      name,
+      description,
+      cash,
+      currency,
+    });
+    return res.status(httpStatus.OK).json(updatedPortfolio);
+  }
+
+  async deletePortfolio(req, res) {
```
Comment (pouyapourbaba) [inline src/controllers/portfolio/portfolio-controller.js:62]:
```suggestion
  async delete(req, res) {
```

Comment (msm-faraj) [reply src/controllers/portfolio/portfolio-controller.js:62]: done

File: src/controllers/portfolio/portfolio-controller.js (hunk 4 above)
Comment (msm-faraj) [reply src/controllers/portfolio/portfolio-controller.js:44]: done

File: src/controllers/portfolio/portfolio-controller.js (hunk 3 above)
Comment (msm-faraj) [reply src/controllers/portfolio/portfolio-controller.js:35]: done

File: src/controllers/portfolio/portfolio-controller.js (hunk 2 above)
Comment (msm-faraj) [reply src/controllers/portfolio/portfolio-controller.js:28]: done

File: src/controllers/portfolio/portfolio-controller.js (hunk 1 above)
Comment (msm-faraj) [reply src/controllers/portfolio/portfolio-controller.js:10]: done

**Review by pouyapourbaba — APPROVED** [review]

**Pull request merged**